### DISCUSSION TITLE
Dynamically file streaming

### DISCUSSION THREAD

**Discussion by Ninos** [ipxe/ipxe#891] [General]
Currently it seems not possible to stream files while using `sanboot` without static file serving (from webserver instead of application/proxying).

What I already tried:
- Attachment disposition inline & attachment
- Set `Accept-Ranges` to `no`
- Set `X-Accel-Buffering` to `no`
- Use file headers (so having correct Content-Length, Type, ...)
- Don't send chunks without content

Can someone tell me what happens exactly while downloading file before sanboot, so I may understand the restrictions between webserver & application file streaming. Seems something is not working like expected... :/

**Comment by mcb30** [comment]
HTTP SAN booting requires the ability to make range requests.  The bootloader stored on the SAN disk will use the INT 13 or UEFI block device APIs to request the blocks that it wants to read.  iPXE will then issue HTTP range requests to retrieve these blocks.

**Reply by Ninos** [reply]
Awesome, thanks, this helped me out! Adding http 206 status code support & cache for faster request handling solved my problem. Again, thx a lot! :)

@NiKiZe yeah makes more sense 👍

**Comment by NiKiZe** [comment]
Moving to discussion until we can establish that this is an "issue"

Now with the correct reason explained it might have been better to close as "can't fix" sorry about that.
